build.go

```diff
@@ -165,6 +173,45 @@ func Build(
 			logger.Debug.Break()
 		}
 
+		var layers []packit.Layer
+		exists, err := fs.Exists(nugetCache.Path)
+		if exists {
+			if !fs.IsEmptyDir(nugetCache.Path) {
+				layers = append(layers, nugetCache)
+			}
+		}
+		if err != nil {
+			return packit.BuildResult{}, err
+		}
+
+		sbomLayer, err := context.Layers.Get("publish")
```

Comment:
I don't think it's necessary to create a separate layer just for storing the SBOM. The Build SBOM without layer association seems like a more natural fit for this data.  Just as the dotnet-execute buildpack [attaches the SBOM to the `BuildResult.Launch.SBOM`](https://github.com/paketo-buildpacks/dotnet-execute/blob/c4cfcd1a6da310c1af08bf8c0a5e2260e50168ab/build.go#L164), this buildpack can attach its SBOM to the [`BuildResult.Build.SBOM`](https://github.com/paketo-buildpacks/packit/blob/18202009038b0df285ba0fb7d8b43abbf60d3ed0/build_metadata.go#L13).

Reply:
I've made this change in my most recent commit.